walkthroughs/howto-k8s-ingress-gateway/v1beta2/manifest.yaml.template

```diff
@@ -0,0 +1,626 @@
+          env:
+            - name: "COLOR"
+              value: "red"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-yellow
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+  selector:
+    app: color
+    version: yellow
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: yellow
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: color
+      version: yellow
+  template:
+    metadata:
+      labels:
+        app: color
+        version: yellow
+      annotations:
+        appmesh.k8s.aws/preview: ${APPMESH_PREVIEW}
+    spec:
+      containers:
+        - name: app
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "COLOR"
+              value: "yellow"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-white
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+  selector:
+    app: color
+    version: white
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: white
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: color
+      version: white
+  template:
+    metadata:
+      labels:
+        app: color
+        version: white
+      annotations:
+        appmesh.k8s.aws/preview: ${APPMESH_PREVIEW}
+    spec:
+      containers:
+        - name: app
+          image: ${COLOR_APP_IMAGE}
+          ports:
+            - containerPort: 8080
+          env:
+            - name: "COLOR"
+              value: "white"
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-headers
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: color-paths
+  namespace: ${APP_NAMESPACE}
+spec:
+  ports:
+    - port: 8080
+      name: http
+---
+apiVersion: v1
+kind: Service
+metadata:
+  name: ingress-gw
+  namespace: ${APP_NAMESPACE}
+spec:
+  type: LoadBalancer
+  ports:
+    - port: 8088
+      name: http
+  selector:
+    app: ingress-gw
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: ingress-gw
+  namespace: ${APP_NAMESPACE}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: ingress-gw
+  template:
+    metadata:
+      labels:
+        app: ingress-gw
+      annotations:
+        appmesh.k8s.aws/sidecarInjectorWebhook: "disabled"
+    spec:
+      containers:
+        - name: envoy
+          image: ${ENVOY_IMAGE}
+          ports:
+            - containerPort: 9080
+          env:
+            - name: "APPMESH_VIRTUAL_NODE_NAME"
+              value: "mesh/${APP_NAMESPACE}/virtualGateway/ingress-gw_${APP_NAMESPACE}"
+            - name: "APPMESH_PREVIEW"
+              value: "1"
+            - name: "ENVOY_LOG_LEVEL"
+              value: "info"
+            - name: "AWS_REGION"
+              value: "$AWS_DEFAULT_REGION"
```

Comment:
Can't this Envoy be automatically configured?

Reply:
The current version of controller does not auto inject this. We will update this once the controller supports auto config for virtual gateway envoy. When to add this change is being discussed right now

Reply:
I think when should be now. Otherwise folks are going to have to redo their gateway setup again, even after adopting the GA controller.

Reply:
That's a fair point. We will add it. I'll hold this PR until controller supports this